pkg/useragent/useragent.go

```diff
@@ -0,0 +1,61 @@
+// package useragent provides a consistent way to get a user agent for outbound http requests from Grafana Agent.
+// The default User-Agent is `GrafanaAgent/$VERSION($MODE)`
+// Where version is the build version of the agent and MODE is one of "static" or "flow".
+package useragent
```

Comment:
Can we put this in `internal/useragent` instead of `pkg/useragent`? Since we're heading towards 1.0 where we're unlikely to have a large public API (or any public API at all), I'd like to get in the habit of only exposing what absolutely needs to be made public.